src/main/java/org/jabref/gui/FindUnlinkedFilesDialog.java

```diff
@@ -527,6 +535,88 @@ public void stateChanged(ChangeEvent e) {
+     * dialogs tree view. <br>
+     * <br>
+     * The export itself will run in a seperate thread, whilst this dialog will
+     * be showing a progress bar, until the thread has finished its work. <br>
+     * <br>
+     * When the export has finished, the {@link #exportFinishedHandler(java.util.List)} is
+     * invoked.
+     */
+    private void startExport() {
+
+        if (treeModel == null) {
+            return;
+        }
+        setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
+
+        CheckableTreeNode root = (CheckableTreeNode) treeModel.getRoot();
+
+        final List<File> fileList = getFileListFromNode(root);
+
+        if ((fileList == null) || fileList.isEmpty()) {
+            return;
+        }
+
+        progressBarImporting.setVisible(true);
+        labelExportingInfo.setVisible(true);
+        buttonExport.setVisible(false);
+        buttonApply.setVisible(false);
+        buttonClose.setVisible(false);
+        disOrEnableDialog(false);
+
+        labelExportingInfo.setEnabled(true);
+
+        progressBarImporting.setMinimum(0);
+        progressBarImporting.setMaximum(fileList.size());
+        progressBarImporting.setValue(0);
+        progressBarImporting.setString("");
+
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+                .withInitialDirectory(Globals.prefs.get(JabRefPreferences.WORKING_DIRECTORY)).build();
+        DialogService ds = new FXDialogService();
+
+        Optional<Path> exportPath = DefaultTaskExecutor
+                .runInJavaFXThread(() -> ds.showFileSaveDialog(fileDialogConfiguration));
+
+        threadState.set(true);
+        JabRefExecutorService.INSTANCE.execute(() -> {
+            try (BufferedWriter writer =
+                         Files.newBufferedWriter(exportPath.get(), StandardCharsets.UTF_8,
+                                 StandardOpenOption.CREATE)) {
+                int counter = 0;
+                for (File file : fileList) {
+                    writer.write(file.toString() + "\n");
+                    counter++;
+                    progressBarImporting.setValue(counter);
+                    progressBarImporting.setString(Localization.lang("%0 of %1", Integer.toString(counter),
+                            Integer.toString(progressBarImporting.getMaximum())));
+                }
+
+            } catch (IOException e) {
+                LOGGER.warn("IO Error.", e);
+            }
+        });
+
+        exportFinishedHandler();
+    }
+
+    /**
+     */
+    private void exportFinishedHandler() {
+
+        progressBarImporting.setVisible(false);
+        labelExportingInfo.setVisible(false);
+        buttonExport.setVisible(true);
+        buttonApply.setVisible(true);
+        buttonClose.setVisible(true);
+        disOrEnableDialog(true);
+        setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+        frame.getCurrentBasePanel().markBaseChanged();
+    }
+
```

Comment:
If you are on it - please remove the method comment 😇  - or add description. The functionality is not exactly clear by the method name.